parquet-column/src/test/java/org/apache/parquet/column/values/dictionary/TestDictionary.java

```diff
@@ -116,10 +117,11 @@ public void testBinaryDictionaryFallBack() throws IOException {
 
     //Fallbacked to Plain encoding, therefore use PlainValuesReader to read it back
     ValuesReader reader = new BinaryPlainValuesReader();
-    reader.initFromPage(100, cw.getBytes().toByteArray(), 0);
+    reader.initFromPage(100, cw.getBytes().toByteBuffer(), 0);
 
     for (long i = 0; i < 100; i++) {
-      assertEquals(Binary.fromString("str" + i), reader.readBytes());
+      assertEquals(Binary.fromString("str" + i).toStringUsingUTF8(),
+                    reader.readBytes().toStringUsingUTF8());
```

Comment:
If it was for readability we should just update the toString method on Binary so that it will be readable for any test that needs to do comparisons. I'll make the test fail to see if the message is bad and update it accordingly so that it prints a helpful message.